### ISSUE TITLE
Template format error: Unresolved resource dependencies [<function name>LogGroup] in the Resources block of the template.

### ISSUE THREAD

**Issue by ridan** [serverless/serverless#6770]
# Bug Report
I was trying to use the serverless monitoring tools and ran `sls` to connect it to my serverless account. After that whenever I try to deploy, I get this message 

`Template format error: Unresolved resource dependencies [CsGetIndexLogGroup] in the Resources block of the template.`
 
My sls template does not have a Resources block. I tried to add an empty Resources block but that did nothing.

`provider:
  versionFunctions: false
  name: aws
  runtime: nodejs8.10
  stage: ${opt:stage, 'dev'}
  csGetIndex:
    handler: handler.getCurrentIndex
    description: Get everything that is indexed
    events:
      - http:
          path: csgetindex
          method: get
          cors: true
`
not showing other functions or the post will get too long.

**Comment by medikoo** [comment]
>My sls template does not have a Resources block
 
I take this error message comes from AWS, so it in general means, that there's a `CsGetIndexLogGroup` mentioned in `DependsOn` section, which doesn't appear in CF template.

It's hard to say what's the source of the issue, just by above information. Are you using nested stacks? Or some other plugins?

**Comment by ridan** [comment]
@medikoo yes I didn't mention it but I am using these serverless plugins: serverless-domain-manager and serverless-nested-stack

**Comment by medikoo** [comment]
@ridan it's definitely related to `serverless-nested-stack`, which most likely puts resource away from main stack (hence is not reachable). Not sure though if it can be related in any way to our monitoring setup, @dschep any thoughts?

**Comment by dschep** [comment]
Could you share your cli output when you run `SLS_DEBUG=t sls deploy`?

**Comment by ridan** [comment]
[output.txt](https://github.com/serverless/serverless/files/3684244/output.txt)
@dschep here you go

**Comment by medikoo** [comment]
@ridan this might be an issue of `serverless-nested-stack` plugin, as from your log output it's clear, that within Logs stack it creates, created LogGroup resources are not exposed on outputs.

And error says that some resource in API stack (as created by `serverless-nested-stack`) cannot access one of the log groups which are contained in Logs stack

**Comment by ridan** [comment]
@medikoo Thank you for pointing that out. Would you by chance know what I can do to solve this? I posted the issue on the plugin github page but it seems like there is no activity on it

**Comment by medikoo** [comment]
> Would you by chance know what I can do to solve this?

You may also try [serverless-plugin-split-stacks](https://github.com/dougmoscrop/serverless-plugin-split-stacks#readme) at least that one worked really well for me (I've also contributed there some options)
